packages/web3-eth/src/utils/watch_transaction_by_subscription.ts

```diff
@@ -0,0 +1,113 @@
+/*
+This file is part of web3.js.
+
+web3.js is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+web3.js is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
+*/
+import { Bytes, Numbers, BlockHeaderOutput, TransactionReceipt } from 'web3-types';
+import { DataFormat, format } from 'web3-utils';
+
+import { NewHeadsSubscription } from '../web3_subscriptions';
+import { transactionReceiptSchema } from '../schemas';
+import { WaitProps, watchTransactionByPolling } from './watch_transaction_by_pooling';
+
+/**
+ * This function watches a Transaction by subscribing to new heads.
+ * It is used by `watchTransactionForConfirmations`, in case the provider supports subscription.
+ */
+export const watchTransactionBySubscription = <
+	ReturnFormat extends DataFormat,
+	ResolveType = TransactionReceipt,
+>({
+	web3Context,
+	transactionReceipt,
+	transactionPromiEvent,
+	returnFormat,
+}: WaitProps<ReturnFormat, ResolveType>) => {
```

Comment:
some of these lines are not covered so patch is 50% in code cov. If its moved functionality wasn't it covered in past?

Reply:
Yes, it is just moved from another file. I separate it to ease the testing of other functions. And it was not covered in the past in the unit tests. Actually, it is only, but well-tested, in the integration test.